### PR TITLE
Regex, Requests for different type of user, request for waiting for approval, aprooving 

### PR DESCRIPTION
**Pull request by 2HEADSs**
1. Add propper RegExp
2. Add isApproved field in User model
3. Add functionality to approve users with "organizer" role;
4. Add routes for:
- awaiting approve
- all regular
- all organizer
- all admin

### CODE REVIEW

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:14]: typo

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:164]: why not in the try catch?

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:207]: Space after "//" and before the next "/"

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:211]: but what if admin and isDeleted === true ?

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:228]: Same question here

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:230]: More explicit error messages

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:226]: "organizerForApprove" -> Is this to get a single "organizer" as the route name states?

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:242]: all

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:219]: Don't forget "isDeleted"

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:276]: Why 24h ? Make it like 5-10 days

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:230]: Isn't this returning all users, not only organizers ?

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:14]: fixed

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:164]: When is in try-catch the error is:
"error": "Cannot read properties of undefined (reading '_id')";
when using "?"operator the error is:
"error":"User not found"
Not sure which one is more useful for FE?

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:207]: fixed

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:226]: typo mistake - for all waiting for approval
Fixed

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:212]: fixed for not deleted.

Comment (2HEADSs) [reply src/controllers/userController.js:212]: throw new Error('You do not have rights to modify the record!');

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:229]: add validation for isDeleted;
throw new Error('You do not have rights to modify the record!'); - if is not admin

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:243]: add allUser fn

File: src/services/userService.js
Comment (2HEADSs) [reply src/services/userService.js:219]: fixed

File: src/services/userService.js
Comment (2HEADSs) [reply src/services/userService.js]: fixed

File: src/services/userService.js
Comment (2HEADSs) [reply src/services/userService.js:291]: fixed with seconds. In documentation is said to use seconds instead of days.

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:294]: The function must be named "getAllUsers()"

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:220]: As discussed, added it directly in the find function.

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:242]: allOrganizers

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:260]: What about deleted users? Currently you are returning only not deleted organizers, but all users. Be consistent. If you want, ypou can pass a parameter where the front end can explicitly pass a param if they need all users/organizers, etc. even the deleted. If not passed, return only not deleted. And it must be consistent.

File: src/controllers/userController.js
Comment (Getsov) [reply src/controllers/userController.js:212]: Where is the fix?

File: src/controllers/userController.js
Comment (Getsov) [reply src/controllers/userController.js:229]: Where is the fix? And here is not modifying, here we have a GET request.

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js:212]: The fix fore isDeleted == true is in the userService:
    if (existingUser.isDeleted == true) {
        throw new Error('User is deleted!');
    }
    Not sure about :
image
the part with admin

Comment (2HEADSs) [reply src/controllers/userController.js:212]: fixed in userSerice

File: src/controllers/userController.js
Comment (2HEADSs) [reply src/controllers/userController.js]: fixed

File: src/services/userService.js
Comment (2HEADSs) [reply src/services/userService.js:227]: From an error handling perspective, I've written it this way.

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:57]: unnecessary "q" letter at the end

File: src/controllers/userController.js
Comment (Getsov) [reply src/controllers/userController.js:166]: I didn't mention the question mark, keep it, but move the code inside the try catch, as you did in the code above

File: src/services/userService.js
Comment (Getsov) [inline src/services/userService.js:225]: Don't explicit mention "admin rights". It's fine here. As it is below.

File: src/controllers/userController.js
Comment (Getsov) [inline src/controllers/userController.js:20]: wrong typed "ckeckData"

**Review by Getsov — APPROVED** [review]

**Pull request merged**
